app/components/post_draft/quick_actions/emoji_quick_action/index.tsx

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useRef} from 'react';
+import {Platform, StyleSheet} from 'react-native';
+import {KeyboardController} from 'react-native-keyboard-controller';
+import {runOnJS, runOnUI} from 'react-native-reanimated';
+
+import CompassIcon from '@components/compass_icon';
+import TouchableWithFeedback from '@components/touchable_with_feedback';
+import {ICON_SIZE} from '@constants/post_draft';
+import {useKeyboardAnimationContext} from '@context/keyboard_animation';
+import {useTheme} from '@context/theme';
+import {DEFAULT_INPUT_ACCESSORY_HEIGHT} from '@hooks/useInputAccessoryView';
+import {changeOpacity} from '@utils/theme';
+
+type Props = {
+    testID?: string;
+    disabled?: boolean;
+};
+
+const style = StyleSheet.create({
+    icon: {
+        alignItems: 'center',
+        justifyContent: 'center',
+        padding: 10,
+    },
+});
+
+export default function EmojiQuickAction({
+    testID,
+    disabled,
+}: Props) {
+    const theme = useTheme();
+    const {
+        isInputAccessoryViewMode,
+        isTransitioningFromCustomView,
+        keyboardHeight,
+        lastKeyboardHeight,
+        inputAccessoryViewAnimatedHeight,
+        showInputAccessoryView,
+        setShowInputAccessoryView,
+        isKeyboardFullyClosed,
+    } = useKeyboardAnimationContext();
+
+    const showEmojiPicker = useCallback(() => {
+        setShowInputAccessoryView(true);
+    }, [setShowInputAccessoryView]);
+
+    const checkCallbackRef = useRef<(() => void) | null>(null);
+
+    const scheduleKeyboardCheck = useCallback(() => {
```

Comment:
This one seems a bit convoluted, so let me know if I understand the usage right.

Press the button, we wait 50 milliseconds to call `scheduleKeyboardCheck`. Then, right at that moment, we do all these checks, and if the keyboard is not closed, we wait a frame to check again. Is that right?

If so, I wonder also, what would happen if we press the button several times too fast. Would we be checking this several times? Can that create any race condition?

A quick look at the code tells me that it shouldn't create any issue, but still wanted to verify with you.